go/vt/vttablet/onlineddl/executor.go

```diff
@@ -1693,6 +1717,43 @@ func (e *Executor) executeMigration(ctx context.Context, onlineDDL *schema.Onlin
 		return failMigration(err)
 	}
 
+	if onlineDDL.StrategySetting().IsCombineDuplicateDDL() {
+		// This migration runs on some table. Let's see if the last migration to complete on the same table,
+		// has the exact same SQL statement as this one.
+		// If so, and because this migration is flagged with -combine-duplicate-ddl, we implicitly mark it as complete.
+		sameSQLasLastCompletedMigration := func() (same bool, completedUUID string, err error) {
+			pendingUUIDs, err := e.readTablePendingMigrationsUUIDs(ctx, onlineDDL.Table)
+			if err != nil {
+				return false, "", err
+			}
+			if len(pendingUUIDs) > 0 {
+				return false, "", nil
+			}
+			found, completedUUID, err := e.getLastCompletedMigrationOnTable(ctx, onlineDDL.Table)
```

Comment:
Here is where it would possibly be beneficial to name this method as "last successful".

If we interpret "completed" as "completed with any status, e.g. successful, failure, cancelled", then this code would seem to do the wrong thing, since we shouldn't skip the duplicate if the last migration wasn't successful.

Reply:
The state `'complete'` is now set in stone, and it is synonym with "successfully complete" as opposed to `'failed'` or `'cancelled'`. Sorry if terminology is confusing, but there is only one interpretation to `complete` in this context.